components/post_emoji/post_emoji.tsx

```diff
@@ -1,11 +1,24 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {CSSProperties} from 'react';
 import PropTypes from 'prop-types';
 
-export default class PostEmoji extends React.PureComponent {
-    static propTypes = {
+interface PostEmojiProps {
+    name: string;
+    imageUrl: string;
+}
+declare module 'react' {
```

Comment:
Just curious, how come we are redefining HTML attributes within the `react` module?

Reply:
@devinbinnie thanks for the review. Without redefining those, I was basically getting this error within the file. Looking at it closely now, it's the `alt` attribute on the `<span>` that causes the warning. 
```
Type '{ alt: string; className: string; title: string; style: { backgroundImage: string; }; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>'.
  Property 'alt' does not exist on type 'DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>'.
```

Reply:
Ahh I see, `alt` only applies to `HTMLImageElement` type objects. I feel like there's some refactoring that needs to happen here, but we'll leave this for now.